cub/device/device_scan.cuh

```diff
@@ -28,844 +27,1974 @@
-    CUB_RUNTIME_FUNCTION
-    static cudaError_t InclusiveScanByKey(
-        void                  *d_temp_storage,              ///< [in] Device-accessible allocation of temporary storage.  When NULL, the required allocation size is written to \p temp_storage_bytes and no work is done.
-        size_t                &temp_storage_bytes,          ///< [in,out] Reference to size in bytes of \p d_temp_storage allocation
-        KeysInputIteratorT    d_keys_in,                    ///< [in] Random-access input iterator to the input sequence of key items
-        ValuesInputIteratorT  d_values_in,                  ///< [in] Random-access input iterator to the input sequence of value items
-        ValuesOutputIteratorT d_values_out,                 ///< [out] Random-access output iterator to the output sequence of value items
-        ScanOpT               scan_op,                      ///< [in] Binary scan functor
-        int                   num_items,                    ///< [in] Total number of input items (i.e., the length of \p d_keys_in and \p d_values_in)
-        EqualityOpT           equality_op = EqualityOpT(),  ///< [in] Binary functor that defines the equality of keys. Default is cub::Equality().
-        cudaStream_t          stream=0,                     ///< [in] <b>[optional]</b> CUDA stream to launch kernels within.  Default is stream<sub>0</sub>.
-        bool                  debug_synchronous=false)      ///< [in] <b>[optional]</b> Whether or not to synchronize the stream after every kernel launch to check for errors.  May cause significant slowdown.  Default is \p false.
-    {
-        // Signed integer type for global offsets
-        typedef int OffsetT;
-
-        return DispatchScanByKey<
-            KeysInputIteratorT, ValuesInputIteratorT, ValuesOutputIteratorT, EqualityOpT, ScanOpT, NullType, OffsetT>
-        ::Dispatch(
-            d_temp_storage,
-            temp_storage_bytes,
-            d_keys_in,
-            d_values_in,
-            d_values_out,
-            equality_op,
-            scan_op,
-            NullType(),
-            num_items,
-            stream,
-            debug_synchronous);
-    }
-
-    //@}  end member group
+  /******************************************************************//**
+   * \name Exclusive scans
+   *********************************************************************/
+  //@{
+
+  /**
+   * @brief Computes a device-wide exclusive prefix sum. The value of `0` is 
+   *        applied as the initial value, and is assigned to `*d_out`.
+   *
+   * @par
+   * - Supports non-commutative sum operators.
+   * - Results are not deterministic for pseudo-associative operators (e.g.,
+   *   addition of floating-point types). Results for pseudo-associative
+   *   operators may vary from run to run. Additional details can be found in
+   *   the [decoupled look-back] description.
+   * - When the input and output sequences are the same, the scan is 
```

Comment:
This point seems redundant with the following one.

Reply:
Thanks for noticing this!